### DISCUSSION TITLE
[Urgent] I want to host the next.js app as a sub application under a website on IIS Server, but it does not work and shows 404. I appreciated if anyone could help~~ Thanks.

### DISCUSSION THREAD

**Discussion by wells-z** [vercel/next.js#64079] [Help]
### Summary

I want to host the next.js app as a sub application under a website on IIS Server, but it does not work and shows 404, if I host as website on IIS, it works, but as company required, I have to host as sub application, otherwise, we will not consider next.js. I appreciated if anyone could help~~ Thanks.

**Please see [GitHub Repo](https://github.com/wells-z/deploynextapp)**

**My Steps(Using HttpPlatformHandler)**

1. Install Node.js(My version is v21.5.0)
2. Download and Install HttpPlatformHandler v1.2  from [Offical Website](https://www.iis.net/downloads/microsoft/httpplatformhandler).
3. Create app using "npx create-next-app@latest".
4. Execute "npm i".
5. Modify "next.config.mjs" and set basePath as '/AppWithNode1'.
6. Execute "npm run build".
7. Copy .next and node_modules folder, and paste it into a new folder which will be the root of sub application.
8. Config IIS(Install IIS, add website and add application pointed to the folder created in Step 5).
9. Add a new web.config(update Node.exe location, mine is C:\Program Files\nodejs\node.exe ).

**I also tried to use iisnode(Please see [aritcle](https://medium.com/@dipneupane/deploying-nextjs-application-on-windows-iis-server-cabdc22bccf8)), but no luck, same issue.**

**next.config.mjs**
```
/** @type {import('next').NextConfig} */
const nextConfig = {
    basePath: '/AppWithNode1',
};

export default nextConfig;
```

**web.config**
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <handlers>
            <add name="httpPlatformHandler" path="*" verb="*" modules="httpPlatformHandler" resourceType="Unspecified" requireAccess="Script" />
        </handlers>
        <httpPlatform stdoutLogEnabled="true" stdoutLogFile=".\node.log" startupTimeLimit="20" processPath="C:\Program Files\nodejs\node.exe" arguments=".\node_modules\next\dist\bin\next start">
            <environmentVariables>
                <environmentVariable name="PORT" value="%HTTP_PLATFORM_PORT%" />
                <environmentVariable name="NODE_ENV" value="Production" />
            </environmentVariables>
        </httpPlatform>
    </system.webServer>
</configuration>
```

**Comment by yrostova** [comment]
I have the same issue with python FAST API app :( Did you find how to fix that ?

**Reply by yrostova** [reply]
this article from Lextudio helped me
https://docs.lextudio.com/blog/running-next-js-web-apps-on-iis-with-httpplatformhandler/

for FAST API framework I made like this: 

```python
from fastapi import FastAPI
from fastapi.routing import APIRouter

# Initialize the FastAPI app
app = FastAPI(
    docs_url="/PythonApp/docs",  # Set the docs URL to be under the sub-path
    redoc_url="/PythonApp/redoc",  # Optional: Set the ReDoc URL under the sub-path
    openapi_url="/PythonApp/openapi.json"  # Set the OpenAPI JSON URL under the sub-path
)

# Create a router with a prefix
router = APIRouter(prefix="/PythonApp")

@router.get("/")
async def read_root():
    return {"Hello": "World"}

@router.get("/items/{item_id}")
async def read_item(item_id: int):
    return {"item_id": item_id}

# Include the router in the FastAPI app
app.include_router(router)
```
